frontend/pulse3d/src/pages/production-console.js

```diff
@@ -1,22 +1,725 @@
+          );
+        },
+      },
+    ],
+    []
+  );
+
+  return (
+    <>
+      <TableContainer>
+        <TableHeader>Channel Firmware</TableHeader>
+        <Table
+          columns={cols}
+          rowData={fwInfo.channel}
+          defaultSortColumn={"version"}
+          rowSelection={false}
+          setRowSelection={null}
+          enableRowSelection={false}
+          enablePagination={false}
+          enableTopToolbar={false}
+          enableSelectAll={false}
+          enableStickyHeader={false}
+          showColumnFilters={false}
+        />
+      </TableContainer>
+      <ChannelFwEditModal
+        channelFwEdit={channelFwEdit}
+        setChannelFwEdit={setChannelFwEdit}
+        mainFwInfo={fwInfo.main}
+        refreshTables={refreshTables}
+      ></ChannelFwEditModal>
+    </>
+  );
+}
+
+function ChannelFwEditModal({ channelFwEdit, setChannelFwEdit, mainFwInfo, refreshTables }) {
+  const [buttons, setButtons] = useState([]);
+  const [labels, setLabels] = useState([]);
+  const [rowInfo, setRowInfo] = useState({});
+
+  useEffect(() => {
+    setButtons(["Close", "Save"]);
+    setLabels(["Main FW Version"]);
+    setRowInfo({ mainFwVersion: channelFwEdit.main_fw_version });
+  }, [channelFwEdit]);
+
+  const mainFwVersions = mainFwInfo.map((row) => row.version).sort(semverSortFn);
+
+  const handleUpdate = async (idx) => {
+    if (idx === 1) {
+      try {
+        const res = await fetch(
+          `${process.env.NEXT_PUBLIC_MANTARRAY_URL}/firmware/channel/${channelFwEdit.version}`,
+          {
+            method: "PUT",
+            body: JSON.stringify({ main_fw_version: rowInfo.mainFwVersion }),
+          }
+        );
+        if (res.status === 200) {
+          setChannelFwEdit({});
+          refreshTables();
+        } else {
+          throw Error;
+        }
+      } catch (e) {
+        setButtons(["Close"]);
+        setLabels(["An error occurred while updating user.", "Please try again later."]);
+      }
+    } else {
+      setChannelFwEdit({});
+    }
+  };
+
+  return (
+    <>
+      <ModalWidget
+        open={Boolean(channelFwEdit.version)}
+        header={`Edit Channel FW v${channelFwEdit.version}`}
+        labels={labels}
+        buttons={buttons}
+        closeModal={handleUpdate}
+      >
+        <DropDownWidget
+          options={mainFwVersions}
+          initialSelected={mainFwVersions.indexOf(channelFwEdit.main_fw_version)}
+          handleSelection={(idx) => {
+            setRowInfo({
+              ...rowInfo,
+              mainFwVersion: mainFwVersions[idx],
+            });
+          }}
+        />
+      </ModalWidget>
+    </>
+  );
+}
+
+function SerialNumberSection() {
+  return (
+    <TableContainer>
+      <TableHeader>Serial Numbers</TableHeader>
       <SerialNumberTable />
-    </Container>
+    </TableContainer>
+  );
+}
+
+export default function ProductionConsole() {
```

Comment:
Do these all need to go on the same page or can we split them out so it's a bit easier to follow?

Reply:
These are probably fine all being on the same page

Reply:
I mean all the components, can we break them up so it's not such a large file?

Reply:
Yeah I can split it up

Reply:
Actually I think that might cause problems with the styling. I can make a note to break everything up once we find a way to reuse all the styling